frontend/src/pages/register.jsx

```diff
@@ -0,0 +1,130 @@
+import {
+  TextField,
+  Stack,
+  Typography,
+  ThemeProvider,
+  createTheme,
+  Button,
+} from "@mui/material";
+import { useState } from "react";
+import { registerUser } from "../services/api";
+
+const theme = createTheme({
+  typography: {
+    fontFamily: "Segoe UI",
+  },
+  palette: {
+    primary: {
+      main: "#ffbd59", // orange
+      contrastText: "#080357", // dark blue
+    },
+  },
+  components: {
+    MuiStack: {
+      styleOverrides: {
+        root: {
+          justifyContent: "center",
+          alignItems: "center",
+          padding: "1em",
+        },
+      },
+    },
+    MuiButton: {
+      styleOverrides: {
+        outlined: {
+          textTransform: "none",
+          fontFamily: '"Segoe UI"',
+          borderColor: "#080357", // custom outline color
+          borderWidth: "2px",
+          color: "#080357",
+          "&:hover": {
+            borderColor: "#ffbd59", // hover color
+            backgroundColor: "#ffbd59",
+            color: "#ffffff",
+          },
+        },
+      },
+    },
+  },
+});
+
+const Register = () => {
+  const [user, setUser] = useState({
+    name: "",
+    email: "",
+    password: "",
+  });
+  const [error, setError] = useState(false);
+  const [displayMessage, setDisplayMessage] = useState("");
+  const handleName = (e) => {
+    const input = e.target.value;
+    setUser((user) => ({
+      ...user,
+      name: input,
+    }));
+  };
+
+  const handleEmail = (e) => {
+    const input = e.target.value;
+    setUser((user) => ({
+      ...user,
+      email: input,
+    }));
+  };
+
+  const handlePassword = (e) => {
+    const input = e.target.value;
+    setUser((user) => ({
+      ...user,
+      password: input,
+    }));
+  };
+
+  const handleRegister = async () => {
```

Comment:
You need to make sure you are properly passing your input to the backend so that it is able to accept the values. it is working for the most part but I am seeing a lot of `TypeError: Cannot read properties of undefined (reading 'name')` for fields in register and login